documentation/contributing/code/index.markdown

```diff
@@ -150,3 +150,26 @@ You can make a specific clang-tidy build with:
 catkin config --cmake-args -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_CXX_CLANG_TIDY=clang-tidy
 catkin build
 ```
+
+### Exceptions to clang-tidy
+
+In some cases, the warnings generated by clang-tidy may be false-positive, e.g. `performance-unnecessary-copy-initialization` may falsely convert a `non-const` variable to `const`, or cannot be silenced for some reason.
```

Comment:
To be honest, I cannot confirm your observations. All changes suggested by clang-tidy look very good.
I reverted all of your exceptions without any issues. See https://github.com/ros-planning/moveit/pull/1376.

That's why I suggest dropping this paragraph.

Reply:
My first thought was that it may be the difference between clang-tidy 3.9 and 6.0. After I removed 3.9 on my disc, and run 6.0 from `clang-tidy clang-tools`, clang-tidy still converts the non-const `robot_state` to const ref. Then I further cleaned up my disc by running `apt remove clang* && apt autoremove` and install clang-tidy 6.0 again. This warning never came up and the rest warnings match the travis_ci result. Perhaps the previous version of clang related pkgs `libllvm` on my disc brings the difference.

Maybe this info and clang-tidy version should be clarified in the doc. Seems clang-tidy 6.0 do actually have more `performance-*` checks and more warnings with the same check than clang-tidy 3.9.